iroh-net/src/key/encryption.rs

```diff
@@ -0,0 +1,153 @@
+//! The private and public keys of a node.
+
+use std::fmt::Debug;
+
+use anyhow::{anyhow, ensure, Result};
+
+pub(crate) const NONCE_LEN: usize = 24;
+
+fn public_ed_box(key: &ed25519_dalek::VerifyingKey) -> crypto_box::PublicKey {
+    crypto_box::PublicKey::from(key.to_montgomery())
+}
+
+fn secret_ed_box(key: &ed25519_dalek::SigningKey) -> crypto_box::SecretKey {
+    crypto_box::SecretKey::from(key.to_scalar())
+}
```

Comment:
how expensive are these two functions?  maybe leave a note about that in (doc) comments so people understand the tradeoffs made here and what to keep in mind when refactoring in the future

Reply:
they are not super cheap unfortunately..though I need to measure them

Reply:
caching added